### PR TITLE
Rename `main` to `run`

### LINKED ISSUE #130: Rename `main` method to `run` or `enter` (or something else)
Right now, there are two `main`s. One from `@effection/node` and another from `effection`, which is confusing. I think that `main` is the understood term as an entry point to an entire process, so `@effection/node` `main` should stay the same since that really is its purpose.

That means we really ought to come up with a better name for it in `effection` proper. I'm thinking that we should call it `run` or `enter`. It should convey that you are entering _an_ effection entry point, but not necessarily for the entire process.

### PR DESCRIPTION
**Pull request by jnicklas**
Closes #130

### CODE REVIEW

File: packages/effection/test.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+let { main } = require('./dist/effection');
+
+main(function*() {
```
Comment (cowboyd) [inline packages/effection/test.js]: Was this file left in intentionally?

**Review by cowboyd — COMMENTED** [review]
Looks good, barring what I think might be an accidental checkin for `effection/test.ts` (I didn't see a reference to it anywhere else) 👍 

We may want to consider extracting the deprecated function even further into `@frontside/javascript` at some later point.

File: packages/effection/test.js (hunk 1 above)
Comment (cowboyd) [reply packages/effection/test.js]: @jnicklas ping

Comment (jnicklas) [reply packages/effection/test.js]: Nice catch 👌

**Review by cowboyd — APPROVED** [review]
👍

**Pull request merged**